x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/physical/local/PushFiltersToSource.java

```diff
@@ -255,10 +255,10 @@ static boolean canPushToSource(Expression exp, LucenePushdownPredicates lucenePu
             return canPushSpatialFunctionToSource(spatial, lucenePushdownPredicates);
         } else if (exp instanceof StringQueryPredicate) {
             return true;
```

Comment:
unrelated to your changes @afoucret , but @carlosdelest I wonder if this check is still needed? the `:` operator is using the match function, so I think we can remove?

Reply:
I am not sure which condition became unnecessary. 

My understanding is that we can remove this one.

```
     if (exp instanceof StringQueryPredicate) {
          return true;
     } 
```

Can you confirm?

Reply:
That's already removed in Ioana's PR - no need to do anything here 👍